x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -109,14 +111,30 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 
 	switch o.getProvider() {
 	case oAuth2ProviderAzure, oAuth2ProviderDefault:
-		creds := clientcredentials.Config{
-			ClientID:       o.ClientID,
-			ClientSecret:   o.ClientSecret,
-			TokenURL:       o.getTokenURL(),
-			Scopes:         o.Scopes,
-			EndpointParams: o.getEndpointParams(),
+		if o.User != "" || o.Password != "" {
+			conf := &oauth2.Config{
+				ClientID:     o.ClientID,
+				ClientSecret: o.ClientSecret,
```

Comment:
Just doublechecking: are ClientID and ClientSecret used/mandatory here?

Reply:
Yes, they are.